### PR TITLE
Reusable workflow to deploy Python package

### PR DESCRIPTION
**Pull request by lpascal-ledger**

### CODE REVIEW

**Comment by tdejoigny-ledger** [comment]
could be good to update the readme

File: .github/workflows/reusable_ragger_tests.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ jobs:
       - name: Install tests dependencies
         run: |
           sudo apt-get update && sudo apt-get install -y qemu-user-static tesseract-ocr libtesseract-dev
+          pip install -U pip setuptools
           pip install --extra-index-url https://test.pypi.org/simple/ -r ${{ inputs.test_dir }}/requirements.txt
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .github/workflows/reusable_ragger_tests.yml]: Probably a rest of legacy. It does not seem harmful to update, so I moved it to `pypi.org`

**Review by tdejoigny-ledger — DISMISSED** [review]

File: .github/workflows/reusable_pypi_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        CHANGELOG_VERSION=$(grep -Po '(?<=## \[)(\d+\.)+[^\]]' "${{ inputs.package_directory }}/CHANGELOG.md" | head -n 1)
+        if [ "${PACKAGE_VERSION}" == "${CHANGELOG_VERSION}" ]; \
+        then \
+            exit 0; \
+        else \
+            echo "Tag '${PACKAGE_VERSION}' and CHANGELOG '${CHANGELOG_VERSION}' versions mismatch!"; \
+            exit 1; \
+        fi
+
+    - name: Display current status
```
Comment (fbeutin-ledger) [inline .github/workflows/reusable_pypi_deployment.yml:76]: `Current parameters` is better
also add the `publish` var

File: .github/workflows/reusable_pypi_deployment.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      run: |
+        echo "Current status is:"
+        echo "- Stable deployment: ${{ inputs.stable_deployment }}"
+        echo "- Will be deployed on ${{ env.DEPLOYMENT_TARGET }}"
```
Comment (fbeutin-ledger) [inline .github/workflows/reusable_pypi_deployment.yml:80]: `Will be deployed on ...`
Not if `publish` is false, to reword I think

**Review by fbeutin-ledger — APPROVED** [review]
Some wording and small scripts remarks, OK for merge once taken into account

**Pull request merged**